test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
I was unsure at first as well. But this looks consistent to me.

If you are fine with the previous test you accept this behaviour

```javascript
set({}, {a: 1}) //=> {}
```

Then you've already chosen to interpret `{}` as "change no keys". That meaning should be the same when `{}` appears nested to be consistent.

Another angle: The current `set` can _never_ remove properties. Missing properties generally means "no change". If this behavior was what you suggest `set` would get a whole new power. In a way that would be a very drastic deviation.